code/datums/components/sisyphus_awarder.dm

```diff
@@ -0,0 +1,49 @@
+/**
+ * This component awards the sisyphus achievement if you cart a boulder from lavaland to centcom
+ * It's not really reusable but its a component just to encapsulate and destroy the behaviour neatly
+ */
+/datum/component/sisyphus_awarder
+	/// What poor sap is hauling this rock?
+	var/datum/weakref/sisyphus
+
+/datum/component/sisyphus_awarder/Initialize()
+	. = ..()
+	if (!istype(parent, /obj/item/boulder))
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/sisyphus_awarder/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_PICKUP, PROC_REF(on_picked_up))
+
+/datum/component/sisyphus_awarder/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_ITEM_PICKUP, COMSIG_ITEM_DROPPED))
+	var/mob/living/struggler = sisyphus?.resolve()
+	if (!isnull(struggler))
+		UnregisterSignal(struggler, COMSIG_ENTER_AREA)
+
+/// Called when we're picked up, check if we're in the right place to start our epic journey
+/datum/component/sisyphus_awarder/proc/on_picked_up(atom/source, mob/living/the_taker)
+	SIGNAL_HANDLER
+	if (!istype(get_area(the_taker), /area/lavaland))
+		qdel(src)
+		return
+	UnregisterSignal(parent, COMSIG_ITEM_PICKUP)
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, PROC_REF(on_dropped))
+	RegisterSignal(the_taker, COMSIG_ENTER_AREA, PROC_REF(on_bearer_changed_area))
+	sisyphus = WEAKREF(the_taker)
```

Comment:
why is this a weakref? we delete if the item is ever dropped. could do it on moved IG? would be more safe

Reply:
yeah I guess if it's always held it's probably reasonably safe to assume that we'll delete alongside the person holding us
I'll change it to moved and use a regular reference